contracts/modules/PermissionManager/AutomatedBlackList.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.4.24;
+import "./GeneralPermissionManager.sol";
+import "../TransferManager/ITransferManager.sol";
+
+/**
+ * @title AutomateBlacklist
+ * @dev to automate blacklist process
+ */
+contract AutomateBlacklist is GeneralPermissionManager,ITransferManager {
+    
+    /**
+     * @dev structure for the blacklist period
+     */
+
+    struct BlacklistDetails {
+        uint256 startDate;
+        uint256 endDate;
+        uint256 repeats;
+    }
+ 
+     /** 
+     *@dev to check if msg.sender can add new entry
+     */
+    modifier isAllowed() {
+        require(msg.sender == securityToken || checkPermission(msg.sender,address(this),"1"));
+        _;
```

Comment:
Wrong usage of `checkPermission()` here use `withPerm(PERM)` modifier here present in the `Module.sol` contract.